cloudwatch_logs_common/test/log_batch_test.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <cloudwatch_logs_common/utils/log_file_manager.h>
+
+using Aws::CloudWatchLogs::Utils::LogFileManager;
+using namespace Aws::CloudWatchLogs;
+using namespace Aws::FileManagement;
+
+const long ONE_DAY_IN_SEC = 86400000;
+
+class TestStrategy : public DataManagerStrategy {
+public:
+  bool isDataAvailable(){
+    return true;
+  }
+
+  DataToken read(std::string &data) override{
+    if(!logs.empty()){
+        data = logs.back();
+        logs.pop_back();
+    }
+
+    data_token++;
+    return data_token;
+  }
+
+  void write(const std::string &data){
+      logs.push_back(data);
+  }
+
+  void resolve(const DataToken &token, bool is_success){
+    if(is_success && token)
+        return;
+    else
+        return;
+    return;
+  }
+
+  std::vector<std::string> logs;
+
+protected:
+
+  uint64_t data_token = 0;
+
+  /**
+   * Options for how and where to store files, and maximum file sizes.
+   */
+  FileManagerStrategyOptions options_;
+};
+
+class LogBatchTest : public ::testing::Test{
+public:
+  void SetUp() override {
+    test_strategy = std::make_shared<TestStrategy>();
+    file_manager = std::make_unique<LogFileManager>(test_strategy);
+  }
+
+  void TearDown() override {
+    log_data.clear();
+    timestamps.clear();
+  }
+
+  void createLogs(){
+    while(!timestamps.empty()){
+        input_event.SetTimestamp(timestamps.back());
+        input_event.SetMessage("Testing readBatch");
+        log_data.push_back(input_event);
+        timestamps.pop_back();
+    }
+  }
+
+  void validateBatch(){
```

Comment:
this seems like you exclusively use it to validate some specific set of data that does not need to be mutable

```suggestion
  void validateBatch(const std::vector<std::string> & entries) const {
```

then this doesn't depend on being stateful

Reply:
in this case can I just pass in a normal array instead of declaring a vector?